src/components/CreateAutomationModal.tsx

```diff
@@ -1,19 +1,47 @@
-import { useContext } from "react"
-import { ModalContext } from "~/state/ModalContext"
+import { type MouseEvent, useContext } from "react"
+import { z } from "zod";
+import { ModalContext, type ModalContextType } from "~/state/ModalContext"
 
-const CreateAutomationModal = () => {
+const eventTypeSchema = z.array(z.object({
+  id: z.number(),
+  name: z.enum(['Calendar', 'Mail', 'GitHub', '']),
+  description: z.string()
+}))
+
+type eventTypes = z.infer<typeof eventTypeSchema>
 
-  const { setCreateModalOpened, setRulesetModalOpened } = useContext(ModalContext)
+const CreateAutomationModal = () => {
+  
+  const { setCreateModalOpened, setRulesetModalOpened, setEventTrigger } = useContext(ModalContext)
+  
+  const eventTypes: eventTypes = [
+    { id: 1, name: 'Mail', description: 'Example: When an email was receieved...' },
+    { id: 2, name: 'GitHub', description: 'Example: When a PR has been merged to a repository...' },
+    { id: 3, name: 'Calendar', description: 'Example: When a meeting is canceled...' },
+    // event types
+  ];
+  
+  const handleClickedAutomation = (e: MouseEvent<HTMLDivElement>, eventType: eventTypes[0]["name"]) => {
+    const eventTrigger = eventType.toLowerCase() as ModalContextType["eventTrigger"]
```

Comment:
How do you know this cast is safe to make?

Reply:
Casting what specifically? If you are talking about eventTrigger variable, the type cast is safe to make because it is using the same type cast in the ModalContext that will be passed in

```
const [eventTrigger, setEventTrigger] = useState<ModalContextType["eventTrigger"]>("");
```

Reply:
Whenever you write `as`, you're telling the type system that you know better than it does and you create a point of potential unsafety in your code. If you're wrong, then you just lost all the benefits of typescript in this region of your code. So my question is...how do you know for sure that `eventType.toLowerCase()` will be of type `as ModalContextType["eventTrigger"]`. Are you 100% certain? Are you 100% certain your current assumptions will never change in the future somehow and break this code?

Reply:
I believe as of right now, I am certain if and only if the eventTypes is correctly written but I think this can definitely cause a bug later in the future since a lot of this code would change.

Reply:
So I think this actually is a concern for right now. I'll make sure to update this